modules/authentication/src/Authentication.ts

```diff
@@ -1,67 +1,47 @@
-import * as models from './models';
-import { AccessToken, RefreshToken, Token, User } from './models';
+import { ManagedModule, ConfigController } from "@conduitplatform/grpc-sdk";
+
+import path from "path";
+import { isNil } from "lodash";
+import { status } from "@grpc/grpc-js";
+import AppConfigSchema from './config';
 import { AdminHandlers } from './admin/admin';
-import AuthenticationConfigSchema from './config';
-import { isNil } from 'lodash';
-import {
-  ConduitServiceModule,
-  DatabaseProvider,
-  GrpcServer,
-  SetConfigRequest,
-  SetConfigResponse,
-} from '@conduitplatform/grpc-sdk';
-import path from 'path';
 import { AuthenticationRoutes } from './routes/routes';
-import { ConfigController } from './config/Config.controller';
-import { ISignTokenOptions } from './interfaces/ISignTokenOptions';
-import { AuthUtils } from './utils/auth';
-import moment from 'moment';
-import { status } from '@grpc/grpc-js';
-import { TokenType } from './constants/TokenType';
-import { v4 as uuid } from 'uuid';
-import randomToken = AuthUtils.randomToken;
-
-export default class AuthenticationModule extends ConduitServiceModule {
-  private database: DatabaseProvider;
-  private _admin: AdminHandlers;
-  private isRunning: boolean = false;
-  private _router: AuthenticationRoutes;
+import * as models from './models';
+import { ISignTokenOptions } from "./interfaces/ISignTokenOptions";
+import { AuthUtils } from "./utils/auth";
+import { TokenType } from "./constants/TokenType";
+import { v4 as uuid } from "uuid";
+import moment from "moment";
+
+export default class Authentication extends ManagedModule {
+  config = AppConfigSchema;
+  service = {
+    protoPath: path.resolve(__dirname, 'authentication.proto'),
+    protoDescription: 'authentication.Authentication',
+    functions: {
+      setConfig: this.setConfig.bind(this),
+      userLogin: this.userLogin.bind(this),
+      userCreate: this.userCreate.bind(this),
+      changePass: this.changePass.bind(this),
+      userDelete: this.userDelete.bind(this),
+    },
+  };
+  private isRunning: boolean = false; // possible abstraction
+  private adminRouter: AdminHandlers;
+  private userRouter: AuthenticationRoutes;
+
+  constructor(moduleName: string) {
+    super(moduleName);
+  }
 
-  async initialize(servicePort?: string) {
-    this.grpcServer = new GrpcServer(servicePort);
-    this._port = (await this.grpcServer.createNewServer()).toString();
-    await this.grpcServer.addService(
-      path.resolve(__dirname, './authentication.proto'),
-      'authentication.Authentication',
-      {
-        setConfig: this.setConfig.bind(this),
-        userLogin: this.userLogin.bind(this),
-        userCreate: this.userCreate.bind(this),
-        changePass: this.changePass.bind(this),
-        userDelete: this.userDelete.bind(this),
-      }
-    );
-    this.grpcServer.start();
-    console.log('Grpc server is online');
+  async onServerStart() {
+    await this.useDatabase();
```

Comment:
I still don't want that function to exist. If you need to wait for a module, do it using the grpc sdk, no abstractions are needed